assets/js/translator.js

```diff
@@ -730,6 +744,131 @@ function translateDoc() {
     }
 }
 
+window.onpopstate = function () {
+    // TODO: Could do the whole restoreChoices here? But would have to call the right translate() functions anyway
+    if(getURLParam('qP').length > 0) {
```

Comment:
makes sense! this and a few things you mentioned below are covered in persistence.js. I have removed it.